kili/mutations/asset/helpers.py

```diff
@@ -314,3 +348,58 @@ def generate_json_metadata_array(as_frames, fps, nb_files, input_type):
             }
         ] * nb_files
     return json_metadata_array
+
+
+def request_signed_urls(auth: KiliAuth, project_id: str, size: int):
+    """
+    Get upload signed URLs
+    Args:
+        auth: Kili Auth
+        project_id: the project Id
+        size: the amount of upload signed URL to query
+    """
+    payload = {
+        "projectID": project_id,
+        "size": size,
+    }
+    urls_response = auth.client.execute(GQL_CREATE_UPLOAD_BUCKET_SIGNED_URLS, payload)
+    return urls_response["data"]["urls"]
+
+
+def upload_data_via_rest(
+    signed_urls: List[str], data_array: List[str], content_type_array: List[str]
+):
+    """upload data in buckets' signed URL via REST
+    Args:
+        signed_urls: Bucket signed URLs to upload local files to
+        path_array: a list of file paths, json or text to upload
+        content_type: mimetype of the data. It will be infered if not given
+    """
+    responses = []
+    for index, data in tqdm.tqdm(enumerate(data_array), total=len(data_array)):
+        content_type = content_type_array[index]
+        headers = {"Content-type": content_type}
+        url_with_id = signed_urls[index]
+        url_to_use_for_upload = url_with_id.split("&id=")[0]
+        if "blob.core.windows.net" in url_to_use_for_upload:
+            headers["x-ms-blob-type"] = "BlockBlob"
+
+        response = requests.put(url_to_use_for_upload, data=data, headers=headers)
+        if response.status_code >= 300:
+            responses.append("")
+            continue
+        responses.append(url_with_id)
+    return responses
+
+
+def check_if_uploading_local_content(content_array: List[str], json_content_array: Optional[list]):
+    """Determine if all data to append is hosted data or local data.
+    If it is a mix of both, return an error
+    """
+    if json_content_array:
+        return False
+    if all(is_url(content) for content in content_array):
+        return False
+    if all(not is_url(content) for content in content_array):
+        return True
+    raise ValueError("Content to append should either be all hosted file or all local files")
```

Comment:
you can specify an Error like `NonUniformContentHostingError`.

Reply:
Indeed! But with the new MR of maxime, this error and this function will be deleted. I take this in the next ticket that I take: https://linear.app/kili-technology/issue/ML-641/aau-on-the-sdk-when-i-use-append-many-to-dataset-i-see-files-are